compiler/util/src/seq.rs

```diff
@@ -7,7 +7,7 @@ use std::sync::Arc;
 pub struct Seq<T> {
     vec: Option<Arc<Vec<T>>>,
 }
-unsafe impl<T> Send for Seq<T> {}
+unsafe impl<T: Send> Send for Seq<T> {}
```

Comment:
This bound is incorrect. If `Seq<T>: Send`, it can be cloned and some of the clones sent to another thread, so that in effect `Seq` via `Deref<Target=[T]>` enables having `&T` alive on multiple threads. Hence this needs `Send + Sync` just like [the corresponding impl for `Arc`](https://doc.rust-lang.org/stable/std/sync/struct.Arc.html#impl-Send-for-Arc%3CT%3E).